### PR TITLE
Add feature to detect duplicates and invalid YYYY-MM

### LINKED ISSUE #61: Add test cases and checking for MonthsPaid that test for valid dates
Valid YYYY-MM: XXXX-[01-12]

### PR DESCRIPTION
**Pull request by waihin26**
Fixes #61

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 1]
```diff
@@ -65,6 +66,28 @@ public CommandResult execute(Model model) throws CommandException {
...
+            if (!monthPattern.matcher(monthStr).matches()) {
+                throw new CommandException(String.format("Invalid month format: %s. "
+                        + "Month must be in YYYY-MM format, where MM is 01-12.", monthStr));
+            }
+        }
+
+        //Check for duplicate months paid
+        for (MonthPaid monthPaid : monthsPaid) {
+            if (existingMonthsPaid.contains(monthPaid)) {
+                throw new CommandException(String.format("Duplicate month paid: %s", monthPaid));
```
Comment (hooyushing) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:87]: Nit: Can seprate the error message into another variable

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 2]
```diff
@@ -65,6 +66,28 @@ public CommandResult execute(Model model) throws CommandException {
...
+        // Define the pattern for valid YYYY-MM format
+        Pattern monthPattern = Pattern.compile("^\\d{4}-(0[1-9]|1[0-2])$");
+
+        // Check for invalid months paid format
+        for (MonthPaid monthPaid : monthsPaid) {
+            // Strip brackets before validation
+            String monthStr = monthPaid.toString().replaceAll("[\\[\\]]", "");
+            if (!monthPattern.matcher(monthStr).matches()) {
+                throw new CommandException(String.format("Invalid month format: %s. "
+                        + "Month must be in YYYY-MM format, where MM is 01-12.", monthStr));
```
Comment (hooyushing) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:80]: Nit: same as below

**Review by hooyushing — COMMENTED** [review]
Small nits, otherwise LGTM

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java (hunk 1 above)
Comment (waihin26) [reply src/main/java/seedu/address/logic/commands/MarkPaidCommand.java]: Updated

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java (hunk 2 above)
Comment (waihin26) [reply src/main/java/seedu/address/logic/commands/MarkPaidCommand.java]: Updated

**Review by hooyushing — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 3]
```diff
@@ -65,6 +66,30 @@ public CommandResult execute(Model model) throws CommandException {
...
+        // Check for invalid months paid format
+        for (MonthPaid monthPaid : monthsPaid) {
+            // Strip brackets before validation
+            String monthStr = monthPaid.toString().replaceAll("[\\[\\]]", "");
+            if (!monthPattern.matcher(monthStr).matches()) {
+                String invalidMonthsMessage = String.format("Invalid month format: %s. "
+                        + "Month must be in YYYY-MM format, where MM is 01-12.", monthStr);
+                throw new CommandException(invalidMonthsMessage);
+            }
+        }
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:83]: Need to move to MonthPaid, not here

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.regex.Pattern;
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:10]: Regex should not be used in *Command.java files. Convention for this project is to use it only in *CommadParser

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 5]
```diff
@@ -65,6 +66,30 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToMark = lastShownList.get(index.getZeroBased());
+        Set<MonthPaid> existingMonthsPaid = personToMark.getMonthsPaid();
+
+        // Define the pattern for valid YYYY-MM format
+        Pattern monthPattern = Pattern.compile("^\\d{4}-(0[1-9]|1[0-2])$");
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:72]: Similarly regex pattern should be in *CommandParser

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 6]
```diff
@@ -65,6 +66,30 @@ public CommandResult execute(Model model) throws CommandException {
...
+            }
+        }
+
+        //Check for duplicate months paid
+        for (MonthPaid monthPaid : monthsPaid) {
+            if (existingMonthsPaid.contains(monthPaid)) {
+                String duplicateMonthsMessage = String.format("Duplicate month paid: %s", monthPaid);
+                throw new CommandException(duplicateMonthsMessage);
+            }
+        }
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:91]: Don't need to check for duplicates as MonthsPaid is a Set of MonthPaid, and duplicates cannot be added to a Set

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 7]
```diff
@@ -65,6 +66,30 @@ public CommandResult execute(Model model) throws CommandException {
...
         Person personToMark = lastShownList.get(index.getZeroBased());
+        Set<MonthPaid> existingMonthsPaid = personToMark.getMonthsPaid();
+
+        // Define the pattern for valid YYYY-MM format
+        Pattern monthPattern = Pattern.compile("^\\d{4}-(0[1-9]|1[0-2])$");
+
+        // Check for invalid months paid format
+        for (MonthPaid monthPaid : monthsPaid) {
+            // Strip brackets before validation
+            String monthStr = monthPaid.toString().replaceAll("[\\[\\]]", "");
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:77]: Operating on `monthPaid.toString()` is weak slap

File: src/test/java/seedu/address/logic/commands/MarkPaidCommandTest.java [hunk 8]
```diff
@@ -39,6 +40,47 @@ public void execute_markPaidPerson_success() {
...
+        assertCommandFailure(command, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_invalidMonthFormat_throwsCommandException() {
+        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        Index index = INDEX_FIRST_PERSON;
+
+        // Use an invalid month format, e.g., "2024-14"
+        MonthPaid invalidMonthPaid = new MonthPaid("[2024-14]");
```
Comment (sumomomomomo) [inline src/test/java/seedu/address/logic/commands/MarkPaidCommandTest.java:70]: This shouldn't be possible: should be `new MonthPaid("2024-14")`

But looking at the code, was an oversight on my part to not add validation in MonthPaid.java. Will open an issue and add it now

Comment (sumomomomomo) [reply src/test/java/seedu/address/logic/commands/MarkPaidCommandTest.java:70]: Also these tests should be moved to MarkpaidCommandParserTest

**Comment by sumomomomomo** [comment]
Implemented in #111 instead. Will close when #111 is implemented

**Pull request closed without merging**